src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.tsx

```diff
@@ -81,67 +81,101 @@ export class DataSourceSelectable extends React.Component<
     this.setState({ ...this.state, isPopoverOpen: false });
   }
 
+  handleSelectedOption(dataSourceOptions: DataSourceOption[], defaultDataSource: string | null) {
+    const [{ id }] = this.props.selectedOption!;
+    switch (id) {
+      case '':
+        this.setState({
+          ...this.state,
+          dataSourceOptions,
+          selectedOption: [{ id: LocalCluster.id, label: LocalCluster.label, checked: 'on' }],
+          defaultDataSource,
+        });
+        this.props.onSelectedDataSources([LocalCluster]);
+        break;
+      case undefined:
+        this.props.notifications.addWarning(
+          i18n.translate('dataSource.fetchDataSourceError', {
+            defaultMessage: 'Data source id is undefined',
+          })
+        );
+        break;
+      default:
+        if (!dataSourceOptions.find((ds) => ds.id === id)) {
+          this.props.notifications.addWarning(
+            i18n.translate('dataSource.fetchDataSourceError', {
+              defaultMessage: 'Data source with id is not available',
+            })
+          );
+          return;
+        }
+        break;
+    }
```

Comment:
Hi Huy. It seems  not handle if plugins passed in a regular selected options. I only see '', undefined and not found.  Also for not found. should we also return  this.props.onSelectedDataSources([]) in undefined cases;